compass/landice/tests/humboldt/mesh.py

```diff
@@ -0,0 +1,347 @@
+
+    Attributes
+    ----------
+    mesh_type : str
+        The resolution or mesh type of the test case
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        mesh_type : str
+            The resolution or mesh type of the test case
+        """
+        super().__init__(test_case=test_case, name='mesh')
+
+        self.add_output_file(filename='graph.info')
+        self.add_output_file(filename='Humboldt_1to10km.nc')
+        self.add_input_file(
+                filename='humboldt_1km_2020_04_20.epsg3413.icesheetonly.nc',
+                target='humboldt_1km_2020_04_20.epsg3413.icesheetonly.nc',
+                database='')
+        self.add_input_file(filename='Humboldt.geojson',
+                            target='Humboldt.geojson',
+                            database=None, copy=True)
+        self.add_input_file(filename='greenland_8km_2020_04_20.epsg3413.nc',
+                            target='greenland_8km_2020_04_20.epsg3413.nc',
+                            database='')
+
+    # no setup() method is needed
+
+    def run(self):
+        """
+        Run this step of the test case
+       """
+        logger = self.logger
+        try:
+            config = self.config
+            section = config['humboldt']
+        except NameError:
+            print('No config file; skipping')
+
+        print('calling build_cell_wdith')
+        cell_width, x1, y1, geom_points, geom_edges = self.build_cell_width()
+        print('calling build_planar_mesh')
+        build_planar_mesh(cell_width, x1, y1, geom_points,
+                          geom_edges, logger=logger)
+        dsMesh = xarray.open_dataset('base_mesh.nc')
+        print('culling mesh')
+        dsMesh = cull(dsMesh, logger=logger)
+        print('converting to MPAS mesh')
+        dsMesh = convert(dsMesh, logger=logger)
+        print('writing grid_converted.nc')
+        write_netcdf(dsMesh, 'grid_converted.nc')
+        # If no number of levels specified in config file, use 10
+        try:
+            levels = section.get('levels')
+        except NameError:
+            levels = '10'
+        print('calling create_landice_grid_from_generic_MPAS_grid.py')
+        args = ['create_landice_grid_from_generic_MPAS_grid.py',
+                '-i', 'grid_converted.nc',
+                '-o', 'gis_1km_preCull.nc',
+                '-l', levels, '-v', 'glimmer']
+        check_call(args, logger=logger)
+
+        print('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'humboldt_1km_2020_04_20.epsg3413.icesheetonly.nc', '-d',
+                'gis_1km_preCull.nc', '-m', 'b', '-t']
+
+        check_call(args, logger=logger)
+
+        print('calling define_cullMask.py')
+        args = ['define_cullMask.py', '-f',
```

Comment:
Is this step necessary if we are culling to geojson extent below? Or do we want the union of both masks?  If so, please add a comment pointing that out.

Reply:
Testing confirms that this isn't necessary, as xCell and yCell are BFB between meshes created with and without this step. Deleting.

Reply:
Just adding a note here that this could still be useful if you want to cull a certain distance from the ice margin, within the geometry defined by the GeoJSON file. This step will be kept in the Thwaites mesh PR (PR #271), but it's fine to remove it here.

Reply:
I'm ok with leaving it in here to make this a general example.  If you leave it, add some comments explaining when/why this could be useful and that it's otherwise not necessary.

Reply:
Okay, I added this back in commit [8635048](https://github.com/MPAS-Dev/compass/pull/272/commits/8635048e9f664399f5681c23d8a2a2f40b8dd54b), and changed it to cull within 5 km of the ice margin so that it has an effect on the mesh. I added a note that it can safely be removed.

Reply:
Could you add a config option that determines if this step happens or not?  You should not plan to have users look at the code at all, the info they need should come through the documentation and the config options they have available to them.

Reply:
Added cullDistance config option in commit [6bfb52b](https://github.com/MPAS-Dev/compass/pull/272/commits/6bfb52b2aa2e2c948fec1e4e8ff2261162216682).